astropy/stats/bayesian_blocks.py

```diff
@@ -482,8 +492,10 @@ def validate_input(
         sigma: float | ArrayLike | None,
     ) -> tuple[NDArray[float], NDArray[float], NDArray[float]]:
         t, x, sigma = super().validate_input(t, x, sigma)
-        if x is not None and np.any(x % 1 > 0):
-            raise ValueError("x must be integer counts for fitness='events'")
+        if (x is not None) and (np.any(x % 1 > 0) or np.any(x < 0)):
+            raise ValueError(
+                "x must be non-negative integer counts for fitness='events'"
+            )
```

Comment:
Excellent ! Can you also add a test to check that this exception is raised as expected (using `pytest.raises`) ?